selfdrive/thermald/thermald.py

```diff
@@ -360,51 +372,29 @@ def thermald_thread():
     should_start = ignition
 
     # with 2% left, we killall, otherwise the phone will take a long time to boot
-    if should_start and msg.thermal.freeSpace <= 0.02:
-      cloudlog.info("Blocking start: <2 percent memory")
-      should_start = False
-
-    # confirm we have completed training and aren't uninstalling
-    if should_start and not (accepted_terms and (not do_uninstall) and
-      (completed_training or current_branch in ['dashcam', 'dashcam-staging'])):
-      cloudlog.info("Blocking start: Did not complete training or uninstalling")
-      should_start = False
-
-    # check if system time is valid
-    if should_start and not time_valid:
-      cloudlog.info("Blocking start: Invalid system time")
-      should_start = False
-
-    # don't start while taking snapshot
+    should_start = startup_allowed(should_start, (msg.thermal.freeSpace > 0.02), "<2 percent memory")
+
+    should_start = startup_allowed(should_start, (not do_uninstall), "Uninstalling")
+
+    should_start = startup_allowed(should_start,
```

Comment:
Split up in accepted terms and completed training. You can also remove whitespace between these checks.

Reply:
Sounds good, should the dashcam branch check have its own error log?

Reply:
It's 

```python
should_start = startup_allowed(should_start, accepted_terms, "Terms not accepted")
should_start = startup_allowed(should_start, (completed_training or current_branch in ['dashcam', 'dashcam-staging']), "Training not completed")
```